docs/FeatureManagement/Samples/TargetingFilter.tests.json

```diff
@@ -0,0 +1,230 @@
+[
+    {
+        "FriendlyName": "DisabledDefaultRollout",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Aiden"},
+        "IsEnabled": {
+            "Result": "false"
+        },
+        "Variant": {
+            "Result": null
+        },
+        "Description": "Targeting Filter, Aiden is not part of the default rollout."
+    },
+    {
+        "FriendlyName": "EnabledDefaultRollout",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Blossom"},
+        "IsEnabled": {
+            "Result": "true"
+        },
+        "Variant": {
+            "Result": null
+        },
+        "Description": "Targeting Filter, Blossom is part of the default rollout."
+    },
+    {
+        "FriendlyName": "TargetedUser",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Alice"},
+        "IsEnabled": {
+            "Result": "true"
+        },
+        "Variant": {
+            "Result": null
+        },
+        "Description": "Targeting Filter, Alice is a targeted user."
+    },
+    {
+        "FriendlyName": "TargetedGroup",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Aiden", "groups":["Stage1"]},
+        "IsEnabled": {
+            "Result": "true"
+        },
+        "Variant": {
+            "Result": null
+        },
+        "Description": "Targeting Filter, Aiden is now targeted because Stage1 is 100% rolled out."
+    },
+    {
+        "FriendlyName": "DisabledTargetedGroup",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"groups":["Stage2"]},
+        "IsEnabled": {
+            "Result": "false"
+        },
+        "Variant": {
+            "Result": null
+        },
+        "Description": "empty/no user will hit the 50% rollout of group stage 2, so it is targeted."
+    },
+    {
+        "FriendlyName": "EnabledTargetedGroup50",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Aiden", "groups":["Stage2"]},
+        "IsEnabled": {
+            "Result": "true"
+        },
+        "Variant": {
+            "Result": null
+        },
+        "Description": "Targeting Filter, Aiden who is not part of the default rollout is part of the first 50% of Stage 2."
+    },
+    {
+        "FriendlyName": "DisabledTargetedGroup50",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Chad", "groups":["Stage2"]},
+        "IsEnabled": {
+            "Result": "false"
```

Comment:
Following dotnet, I implemented the same `isTargeted` in JS. And the result for this test case should be true. (if I impl it correctly)
"Chad\nComplexTargeting\nStage2": ~33.8% < 50%

Inviting @zhiyuanliang-ms to confirm...

Reply:
I validated this in .NET FM. This result should be true.

Reply:
@eskibear @zhiyuanliang-ms, which version are you on? This should be false. You might have different results based on your CPU architecture in the released version.

Reply:
Thanks, then we probably want to use another userId which is >= 50% and be evaluated as false. 
E.g. `Cad`?
 - "Cad\nComplexTargeting\nStage2": ~80.3%

Reply:
? I want this test to return false.
(Pdb) context_id
'Chad\nStage2\nComplexTargeting\nStage2'
(Pdb) percentage
51.995079324579585

Reply:
> Thanks, then we probably want to use another userId which is >= 50% and be evaluated as false. E.g. `Cad`?
> 
> * "Cad\nComplexTargeting\nStage2": ~80.3%

The given inputs **should** return false.

Reply:
I think the context id should be "Chad\nComplexTargeting\nStage2" ?

Reply:
> which version are you on? This should be false. You might have different results based on your CPU architecture in the released version.

It used the little endianness, the same as python.

Reply:
I think there is a bug in python. Hard catch if we didn't do this as we never checked the results in the past.